src/phoebus_driver.cpp

```diff
@@ -1014,11 +1017,211 @@ TaskListStatus PhoebusDriver::MonteCarloStep() {
+            v(0, p::density(), k, j, i), v(0, p::temperature(), k, j, i), lambda);
+        const Real sie = eos.InternalEnergyFromDensityTemperature(
+            v(0, p::density(), k, j, i), v(0, p::temperature(), k, j, i), lambda);
+        const Real h = sie + p / v(0, p::density(), k, j, i) + 1;
+        const Real cs = std::sqrt(bmod / v(0, p::density(), k, j, i) / h);
+        Real divv;
+        Real gam[3][3];
+        Real gam1[3][3];
+        Real gam2[3][3];
+        Real gam3[3][3];
+        Real gam12[3][3];
+        Real gam13[3][3];
+        Real gam23[3][3];
+        Real gam123[3][3];
+        const Real vp[3] = {v(0, p::velocity(0), k, j, i), v(0, p::velocity(1), k, j, i),
+                            v(0, p::velocity(2), k, j, i)};
+        const Real vp1[3] = {v(0, p::velocity(0), k, j, i - 1),
+                             v(0, p::velocity(1), k, j, i - 1),
+                             v(0, p::velocity(2), k, j, i - 1)};
+        geom.Metric(CellLocation::Cent, 0, k, j, i, gam);
+        geom.Metric(CellLocation::Cent, 0, k, j, i - 1, gam1);
+        Real gdet = geom.DetGamma(CellLocation::Cent, 0, k, j, i);
+        const Real W = phoebus::GetLorentzFactor(vp, gam);
+        const Real W1 = phoebus::GetLorentzFactor(vp1, gam1);
+
+        if (ndim == 1) {
+          divv = 1 / gdet *
+                 (geom.DetGamma(CellLocation::Cent, 0, k, j, i) *
+                      v(0, p::velocity(0), k, j, i) / W -
+                  geom.DetGamma(CellLocation::Cent, 0, k, j, i - 1) *
+                      v(0, p::velocity(0), k, j, i - 1) / W1) /
+                 coords.CellWidthFA(X1DIR, k, j, i);
+        }
+
+        if (ndim == 2) {
+          Real vp2[3] = {v(0, p::velocity(0), k, j - 1, i),
+                         v(0, p::velocity(1), k, j - 1, i),
+                         v(0, p::velocity(2), k, j - 1, i)};
+          Real vp12[3] = {v(0, p::velocity(0), k, j - 1, i - 1),
+                          v(0, p::velocity(1), k, j - 1, i - 1),
+                          v(0, p::velocity(2), k, j - 1, i - 1)};
+          geom.Metric(CellLocation::Cent, 0, k, j - 1, i, gam2);
+          geom.Metric(CellLocation::Cent, 0, k, j - 1, i - 1, gam12);
+          Real W2 = phoebus::GetLorentzFactor(vp2, gam2);
+          Real W12 = phoebus::GetLorentzFactor(vp12, gam12);
+
+          divv = 1 / gdet *
+                     (geom.DetGamma(CellLocation::Cent, 0, k, j, i) *
+                          v(0, p::velocity(0), k, j, i) / W -
+                      geom.DetGamma(CellLocation::Cent, 0, k, j, i - 1) *
+                          v(0, p::velocity(0), k, j, i - 1) / W1 +
+                      geom.DetGamma(CellLocation::Cent, 0, k, j - 1, i) *
+                          v(0, p::velocity(0), k, j - 1, i) / W2 -
+                      geom.DetGamma(CellLocation::Cent, 0, k, j - 1, i - 1) *
+                          v(0, p::velocity(0), k, j - 1, i - 1) / W12) /
+                     coords.CellWidthFA(X1DIR, k, j, i) / 2
```

Comment:
since you're doing backwards differences, not centered differences, no 2 here.

Reply:
2 is because I am averaging between j and j-1 coordinate cells for example

Reply:
Yes but that's not how the finite difference formula works. You're taking a derivative. The backwards difference formula is
$df/dx \approx \frac{f(x) - f(x - h)}{h}$
where here $h$ is one factor of the cell width. If you did centered difference, where you averaged forward and backwards differences, you'd have a 2:
$df/dx \approx \frac{f(x+h) - f(x-h)}{2h}$